src/test/regress/spec/isolation_citus_pause_node.spec

```
@@ -0,0 +1,219 @@
+		perform pg_catalog.citus_pause_node_within_txn(v_node_id) ;
+	END;
+	$$
+	LANGUAGE plpgsql;
+}
+
+step "s1-pause-node-force"
+{
+	SET client_min_messages = 'notice';
+	DO $$
+	DECLARE
+		v_shard_id int;
+		v_node_id int;
+		v_node_name text;
+		v_node_port int;
+		v_force boolean := true;
+	BEGIN
+		--The first message in the block is being printed on the top of the code block. So adding a dummy message
+		--to make sure that the first message is printed in correct place.
+		raise notice '';
+		-- Get the shard id for the distribution column
+		SELECT get_shard_id_for_distribution_column('employee', 3) into v_shard_id;
+
+		--Get the node id for the shard id
+		SELECT nodename,nodeport into v_node_name,v_node_port FROM citus_shards WHERE shardid = v_shard_id limit 1;
+		raise notice 'node name is %',v_node_name;
+		raise notice 'node port is %',v_node_port;
+
+		-- Get the node id for the shard id
+		SELECT nodeid into v_node_id FROM pg_dist_node WHERE nodename = v_node_name and nodeport = v_node_port limit 1;
+
+
+		-- Pause the node with force true
+		perform pg_catalog.citus_pause_node_within_txn(v_node_id,v_force) ;
+	END;
+	$$
+	LANGUAGE plpgsql;
+}
+
+step "s1-end"
+{
+    COMMIT;
+}
+
+session "s2"
+
+
+step "s2-begin"
+{
+	BEGIN;
+}
+
+step "s2-insert-distributed"
+{
+	-- Set statement_timeout for the session (in milliseconds)
+	SET client_min_messages = 'notice';
+
+	-- Variable to track if the INSERT statement was successful
+	DO $$
+	BEGIN
+
+		-- Execute the INSERT statement
+		insert into employee values(11,'e11',3);
+
+	END;
+
+	$$
+	LANGUAGE plpgsql;
+}
+
+step "s2-insert-reference"{
+	-- Set statement_timeout for the session (in milliseconds)
+	SET statement_timeout = 1000; -- 1 seconds
+	SET client_min_messages = 'notice';
+
+	-- Variable to track if the INSERT statement was successful
+	DO $$
+	BEGIN
+
+		-- Execute the INSERT statement
+		insert into city values(3,'city3');
+
+	END;
+	$$
+	LANGUAGE plpgsql;
+}
+
+step "s2-select-distributed"{
+	select * from employee where id = 10;
+}
+
+
+step "s2-delete-distributed"{
+	-- Set statement_timeout for the session (in milliseconds)
+	SET client_min_messages = 'notice';
+
+	-- Variable to track if the DELETE statement was successful
+	DO $$
+	BEGIN
+
+		-- Execute the DELETE statement
+		delete from employee where id = 9;
+	END;
+	$$
+	LANGUAGE plpgsql;
+}
+
+step "s2-end"
+{
+	COMMIT;
+}
+
+permutation "s1-begin" "s2-begin" "s1-pause-node" "s2-insert-distributed" "s1-end" "s2-end"
+permutation "s1-begin" "s2-begin" "s1-pause-node" "s2-delete-distributed" "s1-end" "s2-end"
+permutation "s1-begin"  "s1-pause-node" "s2-begin" "s2-select-distributed" "s1-end" "s2-end"
+permutation "s1-begin"  "s2-begin" "s1-pause-node"  "s2-insert-reference" "s1-end" "s2-end"
+permutation "s1-begin"  "s1-pause-node" "s1-pause-node" "s1-end"
+permutation "s1-begin"  "s1-node-not-found" "s1-end"
+permutation "s1-begin"  "s2-begin" "s2-insert-distributed" "s1-pause-node-force"(*) "s2-end" "s1-end"
```

These should be swapped to test that the "force" argument kills the s2 query
```suggestion
permutation "s1-begin"  "s2-begin" "s2-insert-distributed" "s1-pause-node-force"(*) "s1-end" "s2-end"
```

---

```bash
test isolation_setup              ... ok           31 ms
test isolation_cluster_management ... ok           77 ms
test isolation_citus_pause_node   ... FAILED    11001 ms
```
In that case it takes 10 seconds to test the case. If we use so, I need to give a timeout; e.g. 100 ms

---

Used lock_cooldown parameter with the suggestion of @JelteF and it worked